src/main/java/Weeny/Ui.java

```diff
@@ -0,0 +1,134 @@
+package weeny;
+
+import java.util.List;
+import java.util.Scanner;
+
+/**
+ * Handles user interactions by displaying messages and reading input.
+ */
+public class Ui {
+
+    /**
+     * Displays a welcome message to the user.
+     */
+    public void showWelcomeMessage() {
+        printLine();
+        System.out.println("Hello! I'm Weeny\nWhat can I do for you?");
+        printLine();
+    }
+
+    /**
+     * Displays the list of tasks to the user.
+     *
+     * @param tasks The list of tasks to display.
+     */
+    public void showTaskList(List<Task> tasks) {
+        printLine();
+        System.out.println("Here are the tasks in your list:");
+        for (int i = 0; i < tasks.size(); i++) {
+            System.out.println((i + 1) + ". " + tasks.get(i).toString());
+        }
+        printLine();
+    }
+
+    /**
+     * Diplays the list of tasks after search
+     *
+     * @param tasks A list of tasks that contains the keyword
+     */
+    public void showSearchResult(List<Task> tasks) {
+        printLine();
+        System.out.println("Here are the matching tasks in your list:");
+        for (int i = 0; i < tasks.size(); i++) {
+            System.out.println((i + 1) + ". " + tasks.get(i).toString());
+        }
+        printLine();
+    }
+
+    /**
+     * Displays a message indicating a task has been added.
+     *
+     * @param task The task that was added.
+     * @param size The updated number of tasks.
+     */
+    public void printTaskAddedMessage(Task task, int size) {
+        System.out.println("Got it. I've added this task:");
+        System.out.println(task.toString());
+        System.out.println("Now you have " + size + " tasks in the list.");
+    }
+
+    /**
+     * Displays a message indicating a task has been deleted.
+     *
+     * @param task The task that was deleted.
+     * @param size The updated number of tasks.
+     */
+    public void showTaskDeletedMessage(Task task, int size) {
```

Comment:
there seems to be a inconsistency in naming, the previous function was "print" but this one appears to be "show". Maybe they could be similar for consistency?